runtime/js/11_workers.js

```diff
@@ -43,6 +43,9 @@
   }
 
   function decodeMessage(dataIntArray) {
+    if (dataIntArray.length == 0) {
+      return undefined;
+    }
```

Comment:
I don't think this is correct, empty array buffer is a valid message and we shouldn't convert it to `undefined`, it will lead to even more strange problems down the road.

Reply:
@bartlomieju that makes a lot of sense, but currently it does not know how to decode empty array buffer. so what should be returned when we receive empty array?

current behavior throws an exception.

Reply:
Our messaging currently works by having `JSON.stringify()` and `JSON.parse()` on each end, therefore an empty buffer should never happen. However, these `JSON` functions can't be used to transfer `undefined`. Transferring `postMessage(undefined) -> msgEvent.data == undefined` is valid in browsers, which I guess has proper structured cloning. Using empty array values to represent `undefined` is a valid temporary fix IMO. This is implicitly happening on the other end because `encoder.encode(JSON.stringify(undefined)) == new Uint8Array(0)`.

Reply:
@nayeemrmn I agree. 
@bartlomieju what do you think?

Reply:
Sounds like a good temporary solution. @yonbav please add a note above explaining what's going on and mentioning that it's a temporary solution until structured close arrives